src/orng.cpp

```diff
@@ -270,9 +286,95 @@ ACTION orng::setrand(uint64_t job_id, const string& random_value) {
     jobs_table.erase(job_it);
 }
 
+ACTION orng::setranddecen(eosio::name resolver, uint64_t job_id, const string& random_value) {
+    require_auth(resolver);
+    check(!is_paused(), "Contract is paused");
+
+    int64_t running_mode = get_config(running_mode_index, ORACLE_MODE);
+    check(running_mode == DECENTRALIZE_MODE, "RNG is not running decentralize mode");
+
+    node_table.require_find(resolver.value, "Resolver not found, please register first");
+
+    auto job_it = jobs_table.find(job_id);
+    check(job_it != jobs_table.end(), "Could not find job id.");
+
+    uint64_t sig_val{job_it->signing_value};
+
+    auto current_epoch_itr = resolveepoch();
+    auto resolvers = current_epoch_itr->resolvers;
```

Comment:
i am wonder in the epoch race condition happen,  there might be a case that resolvers come from 2 different epochs?

Reply:
> resolvers come from 2 different epochs?

I'm not really understand, can you elaborate this?

Each time only one epoch active with specific choose random resolvers. In any cases that epoch can not pick resolvers (not enough active not ping) then that epoch will be skip.